src/Persistence.LinnApps/Repositories/PartSupplierRepository.cs

```diff
@@ -45,7 +45,8 @@ public IQueryable<PartSupplier> FindAll()
 
         public void Add(PartSupplier entity)
         {
-            throw new NotImplementedException();
+            this.serviceDbContext.Entry(entity.Part).State = EntityState.Unchanged;
```

Comment:
db tries to add a new part Entity instead of tracking the existing without this line. Will try and find a better fix before I merge

Reply:
Feels like there must be a way to include part number in the key without implicating the whole part itself but if not adding a bridgeId wouldn't be the worst thing in the world. 

Also i've made the Add method overridable now so you can just inherit entityframeworkrepository rather then irepository

Reply:
I'll still need all my .Include()'s in this repositorys FindBy() etc methods. Is there some way to get them using the entityframeworkrepository too?

Reply:
yes you can just override the methods that you need to

Reply:
of course! thanks. Also changing the PartRepository call to a FindById() stopped EF core trying to insert a new Part... which makes sense since how else would it know the part in question was actually a part that already existed without knowing the Id 🙃